### DISCUSSION TITLE
v1.0.0

### DISCUSSION THREAD

**Discussion by sigma67** [sigma67/ytmusicapi#375] [Announcements]
It's been 3 years since #10 was created, but now this project finally has a way of OAuth authentication. Therefore it is considered feature complete for now and ready for a v1.0 release, just in time for almost 1k 🌟! The next major release will be v2.0 featuring proper types with pydantic.

Overall there are not many breaking changes in this release, only the setup procedure has changed. It was a priority to integrate the new OAuth authentication as seamlessly as possible.

Features ✨
---------
- support OAuth authentication (#10, thanks @vixalien). 
- `get_search_suggestions`: get the suggestions that appear during typing on YouTube Music (thanks @omgupta15 )
- CLI for setup. Call `ytmusicapi browser` for traditional auth using browser headers or `ytmusicapi oauth` to use the new OAuth authentication (it's simple and fast!)

Breaking Changes ⚒
----------------------
- Setup procedure has been reworked. for library use, you can simple do `from ytmusicapi import setup` instead of importing the `YTMusic` object

Search 🐞
----------------------
- `search` has been a bit problematic since the recent server side changes to top results, it should now be more reliable. If you still notice some odd search results please create an issue

<hr /><em>This discussion was created from the release <a href='https://github.com/sigma67/ytmusicapi/releases/tag/1.0.0'>v1.0.0</a>.</em>

**Comment by vixalien** [comment]
so happy to see this release!

**Reply by Shabinder** [reply]
thanks for your contributions @vixalien 🙌
